### ISSUE TITLE
how to add wrapped solana tokens support

### ISSUE THREAD

**Issue by sergej-js** [fdundjer/solana-sniper-bot#11]
title. I need add wrapped solana tokens suppurt, but i dont know how do it

**Comment by sergej-js** [comment]
In more detail, when specifying the wrapped sol key, everything works, but when it creates a transaction, it writes no data on the sol scan and in the information about the wallet created wallet *TOKEN* 0 tokens. TOKEN_AMOUNT is 0.001.

**Comment by sergej-js** [comment]
![image](https://github.com/fdundjer/solana-sniper-bot/assets/81759686/e9ffbe33-4d2c-441f-9c1d-b4084c1d051d)

**Comment by CVB91** [comment]
I am also having a similar thing occur, I think it might be to do how the amount of SOL is actually getting written into the transaction. Have you tried looking at how the instructions are inscribing the SOL amount into the transactions? I will have another look tonight.

**Comment by sergej-js** [comment]
> I am also having a similar thing occur, I think it might be to do how the amount of SOL is actually getting written into the transaction. Have you tried looking at how the instructions are inscribing the SOL amount into the transactions? I will have another look tonight.

![image](https://github.com/fdundjer/solana-sniper-bot/assets/81759686/669884ce-a9de-4034-8cdb-e7c045afe690)
![image](https://github.com/fdundjer/solana-sniper-bot/assets/81759686/c966738b-8dd4-4343-985f-340fa2690e7a)

**Comment by CVB91** [comment]
This is similar way that I have also set up the script. The issue is that it just isn't sending any wSOL to buy the token, right?

**Comment by sergej-js** [comment]
> This is similar way that I have also set up the script. The issue is that it just isn't sending any wSOL to buy the token, right?

He sends me a log of the purchase. https://solscan.io/tx/2EbPet4iFzC7UCaH8YbFRqzFVFGb8cVqr2ywc2CscrRqQLqtW94XJ2FV9N7a6pGgd3Xq16D6AZZvpfpMDu153S5C?cluster=mainnet-beta
But it's empty, and there's nothing in the wallet either, just a created asscoiatedAddress

**Comment by sergej-js** [comment]
> This is similar way that I have also set up the script. The issue is that it just isn't sending any wSOL to buy the token, right?

![image](https://github.com/fdundjer/solana-sniper-bot/assets/81759686/68a4c346-19ee-468f-9796-a139ae23bacd)
![image](https://github.com/fdundjer/solana-sniper-bot/assets/81759686/970fb8af-dc6c-4a14-a1bb-3a7214f3c89f)

**Comment by CVB91** [comment]
Do you have a telegram or discord we can chat over?

**Comment by fdundjer** [comment]
@CVB91 @sergej-js 
You can write amount in as:
```
const amountIn = new TokenAmount(Token.WSOL, 1, false).raw;
```

For empty transaction issue, you can try changing commitment level to "confirmed".

**Comment by sergej-js** [comment]
> For empty transaction issue, you can try changing commitment level to "confirmed".

Okay i try it. 1 is USDC_TOKEN_AMOUNT?

**Comment by CVB91** [comment]
Thanks. Made the changes but I still only get empty transactions with no token amount bought. Any other ideas as why that could be the case?

**Comment by sergej-js** [comment]
> For empty transaction issue, you can try changing commitment level to "confirmed".

it doesn't work.

**Comment by fdundjer** [comment]
> > For empty transaction issue, you can try changing commitment level to "confirmed".
> 
> it doesn't work.

Try with finalized just to make sure it's not RPC issue. Also you can turn on preflight to see the actual error.

**Comment by rmatu** [comment]
Hey, have anyone been able to set SOL with success?

**Comment by fdundjer** [comment]
@rmatu @sergej-js @Al366io @CVB91 I've updated script to support WSOL. Let me know if it is working as expected.

**Comment by fdundjer** [comment]
Solved with #18
